Themis/Views/Authentication/AuthenticationView.swift

```diff
@@ -121,9 +139,19 @@ struct LoginTextFieldStyle: TextFieldStyle {
             .frame(width: 500, height: 50)
             .background(validInput ? (colorScheme == .light ? Color.black.opacity(0.1) : Color(uiColor: UIColor.systemGray6)) : Color.red)
             .cornerRadius(10)
+            .overlay {
+                if error != nil && type == .username {
+                    RoundedRectangle(cornerRadius: 10)
+                        .stroke(.red.opacity(0.4), lineWidth: 2)
+                }
+            }
     }
 }
 
+enum TextFieldType {
+    case serverURL, username
+}
```

Comment:
Considering that in Swift, the names are not scoped to the file by default and just globally accessible: Maybe we should call this `AuthTextFieldType` to not potentially interfere with later additions to the app?

Reply:
Good point!